File: Sample/CSharp/Sample/Code/TableSpawn.cs
using System;
using System.IO;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using Scorpio.Commons;
using Scorpio.Table;
namespace ScorpioProtoTest {
public class TableSpawn : ITable {
	const string FILE_MD5_CODE = "0ba5325ff1b00e87895cb7961961320a";
    private int m_count = 0;
    private Dictionary<int, DataSpawn> m_dataArray = new Dictionary<int, DataSpawn>();
    public TableSpawn Initialize(TableManager tableManager, string fileName) {
        m_dataArray.Clear();
        ScorpioReader reader = new ScorpioReader(TableUtil.GetBuffer(fileName));
        int iRow = TableUtil.ReadHead(reader, fileName, FILE_MD5_CODE);
        for (int i = 0; i < iRow; ++i) {
            DataSpawn pData = DataSpawn.Read(tableManager, fileName, reader);
            if (Contains(pData.ID()))
                throw new System.Exception("文件[" + fileName + "]有重复项 ID : " + pData.ID());
            m_dataArray.Add(pData.ID(), pData);
        }
        m_count = m_dataArray.Count;
        reader.Close();
        return this;
    }
    public DataSpawn GetElement(int ID) {
        if (Contains(ID)) return m_dataArray[ID];
        TableUtil.Warning("DataSpawn key is not exist " + ID);
        return null;
    }
    public override IData GetValue(int ID) {
        return GetElement(ID);
    }
    public override bool Contains(int ID) {
        return m_dataArray.ContainsKey(ID);
    }
    public override int Count() {
        return m_count;
    }
    public Dictionary<int, DataSpawn> Datas() {
        return m_dataArray;
    }
}
}

File: Sample/Java/Sample/src/ScorpioProtoTest/TableSpawn.java
package ScorpioProtoTest;
import java.nio.ByteBuffer;
import java.nio.ByteOrder;
import java.util.ArrayList;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import Scorpio.Commons.*;
import Scorpio.Table.*;
@SuppressWarnings("unused")
public class TableSpawn extends ITable {
	final String FILE_MD5_CODE = "0ba5325ff1b00e87895cb7961961320a";
    private int m_count = 0;
    private HashMap<Integer, DataSpawn> m_dataArray = new HashMap<Integer, DataSpawn>();
    public TableSpawn Initialize(TableManager tableManager, String fileName) {
        m_dataArray.clear();
        ScorpioReader reader = new ScorpioReader(TableUtil.GetBuffer(fileName));
        int iRow = TableUtil.ReadHead(reader, fileName, FILE_MD5_CODE);
        for (int i = 0; i < iRow; ++i) {
            DataSpawn pData = DataSpawn.Read(tableManager, fileName, reader);
            if (m_dataArray.containsKey(pData.ID()))
                throw new RuntimeException("文件" + fileName + "有重复项 ID : " + pData.ID());
            m_dataArray.put(pData.ID(), pData);
        }
        m_count = m_dataArray.size();
        reader.Close();
        return this;
    }
    public DataSpawn GetElement(Integer ID) {
        if (Contains(ID)) return m_dataArray.get(ID);
        TableUtil.Warning("DataSpawn key is not exist " + ID);
        return null;
    }
    @Override
    public IData GetValue(Integer ID) {
        return GetElement(ID);
    }
    @Override
    public boolean Contains(Integer ID) {
        return m_dataArray.containsKey(ID);
    }
    @Override
    public int Count() {
        return m_count;
    }
    public final HashMap<Integer, DataSpawn> Datas() {
        return m_dataArray;
    }
}
